### PR TITLE
Parse Order and Position entities

### PR DESCRIPTION
**Pull request by lbstr**
More progress on Issue #9 and a continuation of PR #10.

* Parsing numbers and some strings in `Order` and `Position`
* Improved error handling so user code gets a better message if something bad happens in the parse step
* Improving types in a few params interfaces as well

### CODE REVIEW

**Comment by 117** [comment]
Looks like the prettier build step is still failing after updating the `actions/checkout@v2.3.2` version. I'm gonna go ahead and just remove it for now. Can be added later a different way.

File: lib/entities.ts [hunk 1]
```diff
@@ -345,7 +349,316 @@ export interface Order {
   filled_avg_price: string
   status: string
   extended_hours: boolean
-  legs: any
+  legs: RawOrder[]
```
Comment (lbstr) [inline lib/entities.ts:352]: I'm not 100% sure if this works. The docs say that it's either null or a list of orders, but I don't know if the data is really identical. For example, I'm assuming you can only have a depth of 2 -- in other words, your legs can't have legs. If we want to enforce that, I'll need to do a bit more work here...

File: lib/entities.ts [hunk 2]
```diff
@@ -345,7 +349,316 @@ export interface Order {
...
+  trail_price: string
+  trail_percent: string
+  hwm: string
```
Comment (lbstr) [inline lib/entities.ts:355]: These were missing so I added them.

File: lib/parser.ts [hunk 3]
```diff
@@ -1,35 +1,109 @@
...
+        initial_margin: this.parseNumber(rawAccount.initial_margin),
+        maintenance_margin: this.parseNumber(rawAccount.maintenance_margin),
+        last_maintenance_margin: this.parseNumber(
+          rawAccount.last_maintenance_margin
+        ),
+        sma: this.parseNumber(rawAccount.sma),
+        status: rawAccount.status as AccountStatus,
+      }
+    } catch (err) {
+      throw new Error(`Account parsing failed. Error: ${err.message}`)
```
Comment (lbstr) [inline lib/parser.ts:36]: @117 Here's the basic error handling approach. Just try/catch the entire parse step and give a helpful error message. Let me know what you think.

**Comment by 117** [comment]
I'm going to be a bit hands off with reviewing the merges into `dev` and will put a warning on the README.md that it is not recommended for use in a live environment.

**Comment by lbstr** [comment]
@117 That reminds me... I wanted to add something to the README explaining what we're doing here.

File: lib/parser.ts (hunk 3 above)
Comment (117) [reply lib/parser.ts:36]: this is exactly what i had in mind, provides context and the error

**Comment by 117** [comment]
> @117 That reminds me... I wanted to add something to the README explaining what we're doing here.

Like an *About* section?

**Pull request merged**